### PR TITLE
Fix use route node

### LINKED ISSUE #395: [react-router-5] useRouteNode hook is broken
Hello,

Please refer to this [codesandbox](https://codesandbox.io/s/w7y6okrz1w)
I've created two components, one is `ContentWithHocs` and the other is `ContentWithHooks`

From the App component, I call either `ContentWithHocs` or `ContentWithHooks`. It appears that in some case (maybe the first time, before the hook subscribe), the current route returned by `useRouteNode` is null whereas this doesn't show up with the HOC version. 

Maybe the first `useState` should be retrieved from context instead of null ?  

Thanks

### PR DESCRIPTION
**Pull request by titouancreach**
Should fix #395 

Add tests for:
  - `useRouteNode`
  - `routeNode()`

Initialize state with `router.getState()` (the same as the hoc version)

### CODE REVIEW

**Comment by titouancreach** [comment]
I've also fixed name of some tests, and typescript type

File: packages/react-router5/modules/hocs/routeNode.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import RouteNode from '../render/RouteNode'
 
 function routeNode<P>(nodeName: string) {
     return function(BaseComponent: ComponentType<P & RouteContext>): SFC<P> {
-        function RouteNodeWrapper(props) {
+        function RouteNodeWrapper(props: P) {
```
Comment (titouancreach) [inline packages/react-router5/modules/hocs/routeNode.tsx]: I dunno why this is needed but vscode complained. It should have been infered from SFC<P>

**Comment by titouancreach** [comment]
Any update @troch ?

**Comment by troch** [comment]
@titouancreach sorry I haven't had time to look at it yet. Will do this week, and try tonight.

**Comment by titouancreach** [comment]
Thank you very much! :)

**Comment by troch** [comment]
Thanks, sorry for the delay looking into this!

**Pull request merged**